### PR TITLE
✅ Adding test for a simple value modification

### PR DESCRIPTION
**Pull request by gunnarmorling**

### CODE REVIEW

File: src/main/java/io/strimzi/kproxy/interceptor/ProduceRecordMutatingInterceptor.java [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+import io.strimzi.kproxy.codec.DecodedRequestFrame;
+
+/**
+ * An interceptor for modifying the key/value/header/topic of {@link ApiKeys#PRODUCE} requests.
+ */
+public class ProduceRecordMutatingInterceptor implements Interceptor {
+
+    @FunctionalInterface
+    public interface ByteBufferMutation {
+        ByteBuffer mutate(ByteBuffer original);
```
Comment (tombentley) [inline src/main/java/io/strimzi/kproxy/interceptor/ProduceRecordMutatingInterceptor.java:44]: If it is really mutating wouldn't a `void` return type by more obvious?

File: src/test/java/io/strimzi/kproxy/ProxyTest.java [hunk 2]
```diff
@@ -92,10 +95,61 @@ public int port(String host, int port) {
...
+                    @Override
+                    public String host(String host, int port) {
+                        return proxyHost;
+                    }
+
+                    @Override
+                    public int port(String host, int port) {
+                        return proxyPort;
+                    }
+                }),
```
Comment (tombentley) [inline src/test/java/io/strimzi/kproxy/ProxyTest.java:118]: Isn't there some standard address mapping we can use (at least for tests like this one?)

Comment (gunnarmorling) [reply src/test/java/io/strimzi/kproxy/ProxyTest.java:118]: Good point; it was copied from the single original test; I'll try and abstract it.

File: src/main/java/io/strimzi/kproxy/interceptor/ProduceRecordMutatingInterceptor.java (hunk 1 above)
Comment (gunnarmorling) [reply src/main/java/io/strimzi/kproxy/interceptor/ProduceRecordMutatingInterceptor.java:44]: Yeah, it's not mutating in place. Happy about any alternative proposals which don't create that expectation. `modify()`, `convert()`, `alter()`?

File: src/test/java/io/strimzi/kproxy/ProxyTest.java (hunk 2 above)
Comment (gunnarmorling) [reply src/test/java/io/strimzi/kproxy/ProxyTest.java]: Added `FixedAddressMapping` for usage in this test.

**Comment by gunnarmorling** [comment]
Applied; thanks for reviewing, Tom. If you have alternative preferences in terms of naming or anything else, let me know and we can change it.

**Pull request merged**